### PR TITLE
style: Allow passing an nth-index-cache to the invalidation code. 

### PR DESCRIPTION
**Pull request by emilio**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18703)

### CODE REVIEW

File: components/style/invalidation/element/invalidator.rs [hunk 1]
```diff
@@ -923,6 +931,64 @@ impl<'a, 'b: 'a, E> InvalidationCollector<'a, 'b, E>
         );
     }
 
+    /// Check whether a dependency should be taken into account, using a given
+    /// visited handling mode.
+    fn check_dependency(
+        &mut self,
+        visited_handling_mode: VisitedHandlingMode,
+        dependency: &Dependency,
+        relevant_link_found: &mut bool,
```
Comment (bholley) [inline components/style/invalidation/element/invalidator.rs]: Why does this need to be an outparam at all? It doesn't appear to be used in the caller.

File: components/style/invalidation/element/invalidator.rs [hunk 2]
```diff
@@ -997,25 +1031,14 @@ impl<'a, 'b: 'a, E> InvalidationCollector<'a, 'b, E>
...
-                                 None,
-                                 &self.element,
-                                 &mut now_context,
-                                 &mut |_, _| {});
-            if matched_then != matches_now {
+        if is_visited_dependent == VisitedDependent::Yes && relevant_link_found {
+            let should_account_for_dependency = self.check_dependency(
+                VisitedHandlingMode::RelevantLinkVisited,
+                dependency,
+                &mut relevant_link_found,
```
Comment (bholley) [inline components/style/invalidation/element/invalidator.rs]: Oh wait I see, it isn't used here but it is used in the other caller. How about passing a dummy value here instead to make it clearer?

**Comment by bholley** [comment]
r=me with that.

@bor-servo delegate+

**Comment by jdm** [comment]
```
./components/style/data.rs:17: use statement is not in alphabetical order
	expected: selectors::NthIndexCache
	found: selector_parser::{EAGER_PSEUDO_COUNT, PseudoElement, RestyleDamage}

./components/style/invalidation/element/invalidator.rs:19: use statement is not in alphabetical order
	expected: selectors::attr::CaseSensitivity
	found: selectors::NthIndexCache
```

**Pull request merged**